docs/tutorial/esm-limitations.md

```diff
@@ -0,0 +1,25 @@
+# ESM Limitations
+
+This document serves to outline the limitations / differences between ESM in Electron and ESM in Node.js and Chromium.
+
+## Sandboxed preload scripts can't use ESM imports
+
+Sandboxed preload scripts are run as plain javascript without an ESM context.  It is recommended that preload scripts are bundled via something like `webpack` or `vite` for performance reasons regardless, so your preload script should just be a single file that doesn't need to use ESM imports.  Loading the `electron` API is still done via `require('electron')`.
+
+## Non-context-isolated renderers can't use Node.js ESM imports
```

Comment:
I think there's a misunderstanding here (which is good it shows the docs need improvement).  Lemme draw a table and clarify something.

In the renderer process there are two different "ESM" loaders.  Node's and Blinks.  There are also two different ways of doing ESM imports.  The static `import foo from 'bar'` directive and the dynamic `import('bar')` method.  With this information in hand the table below should make more sense

| | NodeJS Static Import | NodeJS Dynamic Import | Blink Static Import | Blink Dynamic Import |
|-|----------------------|----------------------|----------------------|----------------------|
| Sandbox Enabled | No (sandbox means no node) | No (sandbox means no node) | Yes, main world only | Yes, main world only |
| Sandbox Disabled, Context Isolation Enabled | Yes, preload script only | Yes, preload script only | Yes, main world only | Yes, main world only |
| Sandbox Disabled, Context Isolation Disabled | Yes, preload script only | No | Yes, main world only | Yes, main world only |

> if enabling ESM Node imports from preloads requires disabling an important security feature (sandboxing), is it worth supporting at all?

I think the misunderstanding here is that enabling "ESM Node imports" by definitions requires disabling the sandbox because it requires enabling "Node" in the first place.

The only two edge cases (which I tried to document here) are:
* Sandbox off, ctx isolation off: NodeJS dynamic import is not supported anywhere
* Sandbox on: even blinks static and dynamic imports are not supported at all in preload scripts (in the same way that CJS isn't currently supported in these scripts either)